_posts/2022/2022-12-05-find-list-debug-active-record-callbacks-in-the-console.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Fimd out what Callbacks are defined on an Active Record model in the console"
+description: "Sometimes it's hars to know"
+layout: article
+category: ruby
+image:
+  base: "2022/find-list-debug-active-record-callbacks-in-the-console"
+  alt: "Three old payphones"
+  credit: "Pavan Trikutam"
+  source: "https://unsplash.com/photos/71CjSSB83Wo"
+---
+
+I was recently asked by my [First #RubyFriend mentee](https://firstrubyfriend.org) how to list the callbacks present on an Active Record model. I didn't know.
+
+The situation was a large legacy codebase with some large models constructed across multiple files, using various gems, custom callbacks and concerns. It was very hard to parse the callbacks by reading across multiple files.
```

Comment:
*We/They were looking at a vast...with some large models defined (?) across...multiple files. They hooked into a variety of gems and featured custom callbacks and concerns.

Reply:
*parse just the callbacks across the multiple files.

Reply:
```suggestion
They were looking at a vast legacy codebase with some large models defined across multiple files. The models used various gems, custom callbacks and concerns. It was very hard to parse what methods and behaviours the callbacks were causing.
```